cypress/integration/dashboard/dashboard_filtering_spec.js

```diff
@@ -0,0 +1,315 @@
+/**
+ * dashboard_filtering test suite description:
+ * 1) Create a new dashboard, populate with visualizations
+ * 2) Set a filter that excludes all data, and check the visualizations for proper updates
+ * 3) Set the existing filter as pinned, re-check the visualizations
+ * 4) Remove the filter, and check the visualizations for proper updates
+ * 5) Create a new dashboard, and populate with a pie graph
+ * 6) Perform apply different filters on the pie graph and check the visualization for proper updates
+ * 7) Remove all filters and ensure the graph reverts to its original design
+ * 8) Test adding another pie graph to the dashboard and applying a filter to both grpahs
+ */
+
+describe('dashboard filtering', () => {
+  before(() => {
+    // Increase the viewport size to prevent pop-up notifications from blocking UI element
+    // E.g. after adding a saved visualization, a notification appears in the bottom right
+    // which could block clicking on a button located behind it.
+    cy.viewport(1900, 1080)
+
+    // TO DO: Setup the indices
+    // cy.request('PUT', 'localhost:9200/.kibana_1', cy.fixture('dashboard/mappings.json'))
+    // cy.request('POST', 'localhost:9200/.kibana_1', cy.fixture('dashboard/data.json.gz'))
+  })
+
+  after(() => {
+    // TO DO: Tear-down the indices
+    // cy.request('DELETE', 'localhost:9200')
+  })
+
+  describe('adding a filter that excludes all data', () => {
+    before(() => {
+      // Go to the Dashboards list page
+      cy.visit('app/dashboards/list')
+
+      // Click the "Create dashboard" button
+      cy.get('[data-test-subj="newItemButton"]', { timeout: 20000 }).should('be.visible').click()
+      cy.get('[data-test-subj="emptyDashboardWidget"]').should('be.visible')
+      // Change the time to be between Jan 1 2018 and Apr 13, 2018
+      cy.setDashboardDataRange('Apr 13, 2018 @ 00:00:00.000', 'Jan 1, 2018 @ 00:00:00.000')
+
+      // Add all "Filter Bytes Test" visualizations
+      cy.addDashboardPanels('Filter Bytes Test', 'visualization', true)
```

Comment:
where does this `Filter Bytes Test` visualization come from, i didn't see any step to create this visualization. is it suppose covered by load test data which is still todo? if so, make sense.

Reply:
1) The `Filter Bytes Test` visualizations are a group of visualizations that contain "Filter Bytes Test" in the title, and are indeed imported by test data loading functionality (specifically, they are stored in the `cypress/fixtures/dashboard/opensearch_dashboards/data.json.gz`).

2) It may be useful to incorporate the sample data into the visualizations to make it easier to understand, I can look into it in a future PR.